stix2/sdo.py

```diff
@@ -225,12 +321,31 @@ def custom_builder(cls):
 
         class _Custom(cls, _STIXBase):
             _type = type
-            _properties = COMMON_PROPERTIES.copy()
-            _properties.update({
-                'id': IDProperty(_type),
-                'type': TypeProperty(_type),
-            })
-            _properties.update(properties)
+            _properties = OrderedDict()
+            _properties.update([
+                ('type', TypeProperty(_type)),
+                ('id', IDProperty(_type)),
+                ('created_by_ref', ReferenceProperty(type="identity")),
+                ('created', TimestampProperty(default=lambda: NOW, precision='millisecond')),
+                ('modified', TimestampProperty(default=lambda: NOW, precision='millisecond')),
+            ])
+
+            if not properties or not isinstance(properties, list):
+                raise ValueError("Must supply a list, containing tuples. For example, [('property1', IntegerProperty())]")
+
+            _properties.update([x for x in properties if not x[0].startswith("x_")])
+
+            # This is to follow the general properties structure.
+            _properties.update([
+                ('revoked', BooleanProperty()),
+                ('labels', ListProperty(StringProperty)),
+                ('external_references', ListProperty(ExternalReference)),
+                ('object_marking_refs', ListProperty(ReferenceProperty(type="marking-definition"))),
+                ('granular_markings', ListProperty(GranularMarking)),
+            ])
+
+            # Put all custom properties at the bottom, sorted alphabetically.
+            _properties.update(sorted([x for x in properties if x[0].startswith("x_")], key=lambda x: x[0]))
```

Comment:
Since the user is creating a custom SDO instead of adding custom properties to an official SDO, I don't think the non-common properties will start with "x_". (Even custom properties added to official SDOs may not, as it is a SHOULD requirement not a MUST.) Should it check a list of common property names instead?

Reply:
Sure, I could make those changes.

Reply:
I guess that would re-introduce the COMMON_PROPERTIES dictionary and just add properties in the same order they were provided.

Reply:
> custom properties starting with "x_" had to be at the bottom of the object...

Right. But here an entire custom type is being created, so I don't think any properties will start with "x_". My understanding is that "x_" is only if you are adding custom properties to existing, official SDOs.

Reply:
Right. I suppose you could have custom properties on a custom object (if you're extending someone else's custom object), but those should be treated like the custom properties on any other object, and completely separate from the "defined" properties of the custom object itself. I think?